tests/cases/data/source/MongoDbTest.php

```diff
@@ -86,25 +83,99 @@ public function tearDown() {
 		MockMongoPost::reset();
 	}
 
-	public function testBadConnection() {
-		$db = new MongoDb(['host' => null, 'autoConnect' => false]);
-		$this->assertException('Could not connect to the database.', function() use ($db) {
-			$db->connect();
-		});
-		$this->assertTrue($db->disconnect());
-	}
+	public function testConnectNoHost() {
+		$config = $this->_testConfig;
+		$config['host'] = '';
+		$closure = function() use ($config) {
+			$db = new MongoDb($config);
+		};
+		$this->assertException('lithium\core\ConfigException', $closure);
+
+		$config = $this->_testConfig;
+		$config['host'] = null;
+		$closure = function() use ($config) {
+			$db = new MongoDb($config);
+		};
+		$this->assertException('lithium\core\ConfigException', $closure);
+	}
+
+	public function testConnectInvalidDsn() {
+		$config = $this->_testConfig;
+		$config['dsn'] = 'foobar://user:pass@example.org';
+		$closure = function() use ($config) {
+			$db = new MongoDb($config);
+		};
+		$this->assertException('lithium\core\ConfigException', $closure);
+
+		$config = $this->_testConfig;
+		$config['dsn'] = 'foobar://user:pass@example.org/testdb';
+		$closure = function() use ($config) {
+			$db = new MongoDb($config);
+		};
+		$this->assertException('MongoDB\Driver\Exception\InvalidArgumentException', $closure);
+	}
+
+	public function testConnectDsn() {
+		$config = $this->_testConfig;
+		$config['dsn'] = 'mongodb://user:pass@cluster0-shard-00-00-foo.mongodb.net:27017,cluster0-shard-00-01-foo.mongodb.net:27017,cluster0-shard-00-02-foo.mongodb.net:27017/testdb';
+		$db = new MongoDb($config);
+		$this->assertEqual($db->_config['login'], 'user');
+		$this->assertEqual($db->_config['password'], 'pass');
+		$this->assertEqual($db->_config['database'], 'testdb');
+		$this->assertEqual($db->_config['host'], [
+			'cluster0-shard-00-00-foo.mongodb.net:27017',
+			'cluster0-shard-00-01-foo.mongodb.net:27017',
+			'cluster0-shard-00-02-foo.mongodb.net:27017',
+		]);
 
-	public function testGoodConnectionBadDatabase() {
-		$db = new MongoDb(['database' => null, 'autoConnnect' => false]);
+		$config = $this->_testConfig;
+		$config['dsn'] = 'mongodb://cluster0-shard-00-00-foo.mongodb.net:27017,cluster0-shard-00-01-foo.mongodb.net:27017,cluster0-shard-00-02-foo.mongodb.net:27017/testdb';
+		$db = new MongoDb($config);
+		$this->assertEqual($db->_config['database'], 'testdb');
```

Comment:
This looks a little too compact to me. Maybe separate the `assert`s from the initialization/setup phase.